aten/src/ATen/native/AffineGridGenerator.cpp

```diff
@@ -3,24 +3,29 @@
 
 namespace at { namespace native {
 
-at::Tensor linspace_from_neg_one(const Tensor& grid, int64_t num_steps) {
-  if (num_steps > 1) {
-    return at::linspace(-1, 1, num_steps, grid.options());
-  } else {
-    return at::tensor(-1, grid.options());
+at::Tensor linspace_from_neg_one(const Tensor& grid, int64_t num_steps, 
+                                 bool align_corners) {
+  if (num_steps <= 1) {
+    return at::tensor(0, grid.options());
+  }
+  auto range = at::linspace(-1, 1, num_steps, grid.options());
+  if (!align_corners) {
+    range = range * (num_steps - 1) / num_steps;
```

Comment:
can use the helper here as well.

Reply:
This is already itself a helper function, and I think it's essentially as simple as it gets.

If you think that it would make it more maintainable, it could in theory be moved it to `GridSampler.h` (though it's not really a grid sampler function), or it could be moved into its own new `AffineGridGenerator.h` (but it would end up being a header file with just this one function), but I would say it's probably fine as is.

Reply:
Note that it can't make use of the helper functions from the grid sampler or from upsample since it's just not the same the computation. It's converting between different sets of units than those are.